src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -166,14 +169,14 @@ def check_literal(node, dim, index, value):
     range index is valid.
 
     :param array: the node to check.
-    :type array: :py:class:`pysclone.psyir.node.array`
+    :type array: :py:class:`pysclone.psyir.node.Array`
     :param int dim: the dimension index to check.
     :param int index: the index of the range to check (0 is the \
         lower bound, 1 is the upper bound).
     :param int value: the expected value of the literal.
 
-    :raises InternalError: if the supplied argument does not have the \
-    required properties.
+    :raises NotImplementedError: if the supplied argument does not \
+        have the required properties.
 
     '''
     try:
```

Comment:
Again, please document the fact that dim must be >=1 and extend the error handling.

Reply:
OK, done. This required a new function to avoid replication between `_check_bound_is_full_extent` and `_check_literal`